### PR TITLE
Added clean exec functionality + ACI implementation

### PR DESCRIPTION
**Pull request by erjadi**
Clean implementation of exec for Virtual Kubelet:
- Reusing standard Kubelet Executor interface (and remotecommand functions)
- Implemented ACI provider

To be fixed before merge:
- How to pass namespace to provider (currently doing ACI specific way, should be changed)
- Expected streamcount is fixed, but should be based on exec request to apiserver. However when using exec request, createStreams will wait indefinitely (for +1 stream).

### CODE REVIEW

File: providers/azure/client/aci/exec.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+	b := new(bytes.Buffer)
+
+	if err := json.NewEncoder(b).Encode(xc); err != nil {
+		return xcrsp, fmt.Errorf("Encoding create launch exec body request failed: %v", err)
+	}
+
+	req, err := http.NewRequest("POST", uri, b)
+	if err != nil {
+		return xcrsp, fmt.Errorf("Creating get container logs uri request failed: %v", err)
```
Comment (ewbankkit) [inline providers/azure/client/aci/exec.go]: `get container logs` -> `launch exec`

**Comment by rbitia** [comment]
Hey so I'm trying this out rn: 

Rias-MacBook-Pro:demos riabhatia$ kubectl exec -it testnginx -- /bin/bash
Error from server: error dialing backend: dial tcp 10.244.0.10:10250: getsockopt: connection refused

Does it work form cmdline yet?

**Comment by rbitia** [comment]
FYI pushing this VK image with exec to docker hub w/ this name: rbitia/vk:exec

**Comment by rbitia** [comment]
@erjadi Hey Eric can you run a 'dep ensure' and push again?

**Comment by erjadi** [comment]
Hi @rbitia let me do that later tonight and push to this pull request. I'll provide a docker image as well so people could try without needing to build from source.

I did have exec working from command line on my machine, but I'd be interested to know if it works for others as well.

**Comment by erjadi** [comment]
After merging from master and running dep ensure I'm getting a lot of build errors:

`# github.com/virtual-kubelet/virtual-kubelet/vendor/k8s.io/client-go/discovery
vendor/k8s.io/client-go/discovery/restmapper.go:42:75: undefined: meta.VersionInterfacesFunc
vendor/k8s.io/client-go/discovery/restmapper.go:176:19: undefined: meta.VersionInterfacesFunc
# github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/hyperhq/libcompose/config
vendor/github.com/hyperhq/libcompose/config/hash.go:52:23: s.MapParts undefined (type yaml.SliceorMap has no field or method MapParts)
vendor/github.com/hyperhq/libcompose/config/hash.go:58:60: s.MapParts undefined (type yaml.SliceorMap has no field or method MapParts)
vendor/github.com/hyperhq/libcompose/config/hash.go:61:18: s.Slice undefined (type yaml.MaporEqualSlice has no field or method Slice)
vendor/github.com/hyperhq/libcompose/config/hash.go:68:18: s.Slice undefined (type yaml.MaporColonSlice has no field or method Slice)
vendor/github.com/hyperhq/libcompose/config/hash.go:75:18: s.Slice undefined (type yaml.MaporSpaceSlice has no field or method Slice)
vendor/github.com/hyperhq/libcompose/config/hash.go:82:18: s.Slice undefined (type yaml.Command has no field or method Slice)
vendor/github.com/hyperhq/libcompose/config/hash.go:89:18: s.Slice undefined (type yaml.Stringorslice has no field or method Slice)
vendor/github.com/hyperhq/libcompose/config/merge.go:99:23: config.EnvFile.Slice undefined (type yaml.Stringorslice has no field or method Slice)
vendor/github.com/hyperhq/libcompose/config/merge.go:107:28: config.Environment.Slice undefined (type yaml.MaporEqualSlice has no field or method Slice)
vendor/github.com/hyperhq/libcompose/config/merge.go:109:29: config.EnvFile.Slice undefined (type yaml.Stringorslice has no field or method Slice)
vendor/github.com/hyperhq/libcompose/config/merge.go:109:29: too many errors`

Not sure where these come from since I did not touch these libraries. Also I noticed that some of the files in /vendor are now symlinks that seem to be incorrect, for example : https://github.com/virtual-kubelet/virtual-kubelet/blob/master/vendor/k8s.io/kubernetes/.bazelrc

**Comment by rbitia** [comment]
@robbiezhang can you help Eric out?

**Comment by erjadi** [comment]
Closed -> see https://github.com/virtual-kubelet/virtual-kubelet/pull/244

**Pull request closed without merging**